src/components/ListItem.jsx

```diff
@@ -1,5 +1,69 @@
+import Checkbox from '@mui/material/Checkbox';
+import { useState, useEffect } from 'react';
 import './ListItem.css';
+import { updateItem } from '../api';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ item }) {
+	// Destructure item props
+	const { name, dateLastPurchased } = item;
+
+	// Get the initial checked state from localStorage or default to false
+	const [checked, setChecked] = useState(() => {
+		const storedChecked = localStorage.getItem(`checked-${name}`);
+		return storedChecked ? JSON.parse(storedChecked) : false;
+	});
+
+	// Function to check if 24 hours have passed
+	const has24HoursPassed = (dateLastPurchased) => {
```

Comment:
Nice function naming! I'm always a fan of easily readable code!

Reply:
Agree!